### DISCUSSION TITLE
Consolidate Pressure Outlet boundary markers for both compressible and incompressible

### DISCUSSION THREAD

**Discussion by getwelsim** [su2code/SU2#2143] [General]
Hi SU2 developers, 

Based on the user manual at [https://su2code.github.io/docs_v7/Markers-and-BC/#outlet-boundary-condition](https://su2code.github.io/docs_v7/Markers-and-BC/#outlet-boundary-condition), the incompressible pressure outlet boundary condition (BC) needs an additional command 
*INC_OUTLET_TYPE= PRESSURE_OUTLET*. 

Is there any reason to explicitly claim the compressibility for this BC? It seems that the program can tell the applied solver. Consolidating these two pressure outlet BCs can make the command simpler. Thanks.

**Comment by pcarruscag** [comment]
The type can also be "mass flow outlet"

**Reply by lcheng9** [reply]
Thank you for your response @pcarruscag . 
Yes. the "mass flow outlet" is kept. Only consolidate two "pressure outlet" markers.

**Reply by pcarruscag** [reply]
I dont understand what you mean by consolidating, the name of the option is already the same.

**Reply by lcheng9** [reply]
Now I understand the reason for specifying the "INC_OUTLET_TYPE= PRESSURE_OUTLET", because of the mass flow outlet. Thank you for the answer, I can live with the current commands for the pressure outlet.

**Reply by pcarruscag** [reply]
So the issue is that INC_OUTLET_TYPE is a list, it needs one type per outlet marker, so that you can mix pressure and mass flow outlets.
Because it is a list, it does not have default values.
In CConfig::SetPostprocessing we could detect if INC_OUTLET_TYPE is not defined, and fill it with default PRESSURE_OUTLET.

**Reply by lcheng9** [reply]
Make sense, this command design allows mixed outlets. And the default outlet is pressure, even if the INC_OUTLET_TYPE is not specified. Thank you.

**Reply by pcarruscag** [reply]
To be clear that is not implemented, Im suggesting it as something that can be done to get the consistency you mentioned.
